client/my-sites/plans-features-main/index.tsx

```diff
@@ -560,7 +429,68 @@ const PlansFeaturesMain = ( {
 	 * Eventually once the spotlight card is made responsive this flag can be removed.
 	 * Check : https://github.com/Automattic/wp-calypso/pull/80232 for more details.
 	 */
-	const isSpotlightOnCurrentPlanAllowed = SPOTLIGHT_ENABLED_INTENTS.includes( intent );
+	const gridPlanForSpotlight =
+		sitePlanSlug && isSpotlightOnCurrentPlan && SPOTLIGHT_ENABLED_INTENTS.includes( intent )
+			? gridPlansForFeaturesGrid.find(
+					( { planSlug } ) => getPlanClass( planSlug ) === getPlanClass( sitePlanSlug )
+			  )
+			: undefined;
+
+	/**
+	 * Calculates the height of the masterbar if it exists, and passes it to the component as an offset
+	 * for the sticky CTA bar.
+	 */
+	useLayoutEffect( () => {
+		const masterbarElement = document.querySelector< HTMLDivElement >( 'header.masterbar' );
+
+		if ( ! masterbarElement ) {
+			return;
+		}
+
+		if ( ! window.ResizeObserver ) {
+			setMasterbarHeight( masterbarElement.offsetHeight );
+			return;
+		}
+
+		let lastHeight = masterbarElement.offsetHeight;
+
+		const observer = new ResizeObserver(
+			( [ masterbar ]: Parameters< ResizeObserverCallback >[ 0 ] ) => {
+				const currentHeight = masterbar.contentRect.height;
+
+				if ( currentHeight !== lastHeight ) {
+					setMasterbarHeight( currentHeight );
+					lastHeight = currentHeight;
+				}
+			}
+		);
+
+		observer.observe( masterbarElement );
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [] );
+
+	useLayoutEffect( () => {
+		if ( showPlansComparisonGrid ) {
+			setTimeout( () => {
+				if ( plansComparisonGridRef.current ) {
+					scrollIntoViewport( plansComparisonGridRef.current, {
+						behavior: 'smooth',
+						scrollMode: 'if-needed',
+						block: 'nearest',
+						inline: 'nearest',
+					} );
+				}
+			} );
+		}
+	}, [ showPlansComparisonGrid ] );
+
+	useEffect( () => {
+		recordTracksEvent( 'calypso_wp_plans_test_view' );
```

Comment:
@niranjan-uma-shankar It looks like you first introduced this event in https://github.com/Automattic/wp-calypso/commit/406b50c47ab#diff-94b4dba2781ff7ebef1a601f668e6c9849cd03f6f9c78526cdd971b8f98dba88 . Could you help me clarify its purposes and why it is named with a "test" inside? I'm asking because it is not formally registered and its name reads temporary for me, so I'm wondering if it's time to formalize it :)